chapter-1-basics.md

```diff
@@ -317,35 +335,42 @@ thus signal `ReadyToWork` to the ProcessEngine.
 * `ActivityClassification.Production` is used to notify that the cell is ready to
  work on an `Activity` of type `production`.
 
-Since that should result in an `ActivityStarted` event, the next thing to
- implement will be the `ActivityStarted()` function. Find the comment
-`/* Start execution here */` and replace it so that the whole function looks 
-like this:
+Since that should result in an `ActivityStarted` event, the next thing in
+ the implementation is the `ActivityStarted()` function. Find the comment
+`/* Start execution via driver */`. The whole function looks like this:
 
 ```cs
 public override void StartActivity(ActivityStart activityStart)
 {
     _currentSession = activityStart;
     switch (activityStart.Activity)
     {
-        case  AssemblingActivity activity:
-            VisualInstructor.Execute(Name, activityStart, CompleteInstruction);
+        case AssemblingActivity activity:
+            /* Start execution via driver */
+            if (VisualInstructor != null)
+            {
+                VisualInstructor.Execute(Name, activityStart, InstructionCompleted);
+            }
+            else if (Driver != null)
+            {
+                Driver.Output["Start"] = true;
+            }
             break;
     }
 }
 ```
 
-That wouldn't compile so far, because there is no `CompleteInstruction` right 
-now, which is provided as a delegate to `Instructor.Execute`. That means, you 
-have to implement `CompleteInstruction()`, that gets called, when an instruction
-has finished, i.e.: When a worker has finished their task.
+This alone wouldn't compile so far. There also hast to be a method `InstructionCompleted`,
+which is provided as a delegate to `Instructor.Execute`. That means, you need an implemented
+`CompleteInstruction()`, that gets called, when an instruction has finished,
+i.e.: When a worker has finished their task.
 
 ```cs
-private void CompleteInstruction(int instructionResult, ActivityStart activity)
+private void InstructionCompleted(int result, ActivityStart session)
```

Comment:
If this is a callback, this name is ok. In case this method is called throughout the code to *complete instructions*, I'd keep the original version.